File: CWE_Python/src/CWE/CWE20/Complaint.py
#      CWE-20: Improper Input Validation

def quantity():
    return -1  # negative value causes user account credited instead of debited


def charge(total):
    bal = 100
    return bal - total


price = 20
quantity = quantity()
if quantity >= 1:
    total = price * quantity
    print(charge(total))
else:
    print("Invalid Quantity!!")

File: CWE_Java/src/main/java/com/example/cwe/CWE20/Complaint.java
package com.example.cwe.CWE20;

//      CWE-20: Improper Input Validation

public class Complaint {

    public static void main(String[] args){
        double price = 20.00;
        int quantity = getQuantity();

        if(quantity >= 1) {
        double total = price * quantity;
        chargeUser(total);
        }
        else {
            System.out.println("Invalid Quantity!!");
        }

    }

    static void chargeUser(double total) {
        double userBal = 100;
        System.out.println(userBal - total);
    }
    static int getQuantity() {
        return -1;
    }

}
